tests/test_polymorphic.py

```diff
@@ -0,0 +1,32 @@
+from rest_framework import status
+from rest_framework.reverse import reverse
+
+
+def test_get_pets(api_client):
+    url = reverse("pets")
+
+    response = api_client.get(url)
+
+    assert response.status_code == status.HTTP_200_OK
+
+    expected_data = [
+        {"name": "Snowball", "pet_type": "cat", "hunting_skill": "lazy"},
+        {"name": "Lady", "pet_type": "dog", "bark": "soft"},
+        {"name": "John", "pet_type": "lizard", "loves_rocks": True},
+    ]
+
+    assert response.json() == expected_data
+
+
+def test_get_schema(api_client):
```

Comment:
You can also take a look at the drf-spectacular tests, where they include expected `schema.yaml` in the test suite and compare that with the actual retrieved result. That instantly documents the expected schema generation as well.